tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -482,84 +489,216 @@ connect_redirect_test_wrapper(
     delete sender_socket;
 }
 
-void
-connect_redirect_tests_common(_In_ const struct bpf_object* object, bool dual_stack, _In_ test_addresses_t& addresses)
-{
-    // First category is authorize tests.
-    const char* protocol_string = (_globals.protocol == IPPROTO_TCP) ? "TCP" : "UDP";
-    const char* family_string = (_globals.family == AF_INET) ? "IPv4" : "IPv6";
-    const char* dual_stack_string = dual_stack ? "Dual Stack" : "No Dual Stack";
-
-    // Loopback address.
-    printf("CONNECT: Loopback | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
-    authorize_test_wrapper(object, dual_stack, addresses.loopback_address);
-
-    // Remote address.
-    printf("CONNECT: Remote | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
-    authorize_test_wrapper(object, dual_stack, addresses.remote_address);
-
-    // Local non-loopback address.
-    printf("CONNECT: Local | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
-    authorize_test_wrapper(object, dual_stack, addresses.local_address);
-
-    // Second category is connection redirection tests.
-
-    // Remote -> Remote
-    printf("REDIRECT: Remote -> Remote | %s | %s | %s\n", protocol_string, family_string, dual_stack_string);
-    connect_redirect_test_wrapper(object, addresses.vip_address, addresses.remote_address, dual_stack);
+#define DECLARE_CONNECTION_AUTHORIZATION_TEST_FUNCTION(destination)                                              \
```

Comment:
This diff should not be in this PR. You may want to rebase your topic branch.

Reply:
updated